src/EventListener/ReadListener.php

```diff
@@ -33,12 +34,19 @@
     private $collectionDataProvider;
     private $itemDataProvider;
     private $subresourceDataProvider;
+    private $identifierNormalizer;
 
-    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null)
+    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null, IdentifierNormalizerInterface $identifierNormalizer = null)
     {
         $this->collectionDataProvider = $collectionDataProvider;
         $this->itemDataProvider = $itemDataProvider;
         $this->subresourceDataProvider = $subresourceDataProvider;
+
+        if (null === $identifierNormalizer) {
```

Comment:
Is this mandatory, can't we just do nothing if `null` (in many cases, you don't need to normalize the identifier at all).